tools/engine_tool/lib/src/build_utils.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:engine_build_configs/engine_build_configs.dart';
+
+import 'environment.dart';
+
+/// A function that returns true or false when given a [BuildConfig] and its
+/// name.
+typedef ConfigFilter = bool Function(String name, BuildConfig config);
+
+/// A function that returns true or false when given a [BuildConfig] name
+/// and a [GlobalBuild].
+typedef BuildFilter = bool Function(String configName, GlobalBuild build);
+
+/// Returns a filtered copy of [input] filtering out configs where test
+/// returns false.
+Map<String, BuildConfig> filterBuildConfigs(
+    Map<String, BuildConfig> input, ConfigFilter test) {
+  final Map<String, BuildConfig> r = <String, BuildConfig>{};
```

Comment:
This is a great place for the inline-for, imo:

```dart
return {
  for (final entry in input.entries)
    if (test(entry.name, entry.value))
      entry.name: entry.value
};
```

Reply:
Done.